src/utils/UniswapV3Seller.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: CC0-1.0
+
+pragma solidity 0.8.15;
+
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/IPeripheryImmutableState.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/external/IWETH9.sol";
+import "../interfaces/ITokenSeller.sol";
+import "./Sacrifice.sol";
+
+/**
+ * @title UniswapV3Seller
+ * Helper for selling some token for ETH through a 2-hop UniswapV3 exchange.
+ */
+contract UniswapV3Seller is ITokenSeller {
```

Comment:
Does it make sense to provide a method in the ITokenSeller and its successors to estimate how much can be received for a specific amount of BOB tokens? For me it seems reasonable since we hide a specific way how BOBs are being exchanged to the native tokens. The UI will call this method from the pool/token seller to get expected (and minimal) amount of tokens.

Reply:
Sure, it will simplify things for the UI. Added `quoteSellForETH` method in token seller interface.